### PR TITLE
Fix kernel builder c_if qubit extraction bug

### PR DESCRIPTION
**Pull request by anthony-santana**
### Description

* Fixes an issue where qubits that are extracted in a `then` function were not getting a unique `extract_ref` if used outside of the conditional
* Adds new python and C++ tests to test for this bug

### CODE REVIEW

**Review by amccaskey — APPROVED** [review]
Looking at this, I'm pretty sure the code you've removed was put there before we had the canonicalizers that combined like extracts. Good to see the old stuff get cleaned up.

File: runtime/cudaq/builder/QuakeValue.cpp [hunk 1]
```diff
@@ -168,9 +155,10 @@ QuakeValue QuakeValue::operator[](const QuakeValue &idx) {
   if (type.isa<quake::VeqType>()) {
     Value extractedQubit =
         opBuilder.create<quake::ExtractRefOp>(vectorValue, indexVar);
-    auto ret = extractedFromValue.emplace(
-        std::make_pair(opaquePtr, QuakeValue(opBuilder, extractedQubit)));
-    return ret.first->second;
+    // auto ret = extractedFromValue.emplace(
```
Comment (amccaskey) [inline runtime/cudaq/builder/QuakeValue.cpp:158]: should remove the commented code

**Review by schweitzpgi — COMMENTED** [review]
+1

**Pull request merged**